viewing_party/main.py

```diff
@@ -0,0 +1,195 @@
+# (by title) from watchlist to empty watched list
+
+# test 8 moves a movie from watched list to watchlist when watchlist is truthy
+
+# test 9 function does nothing if movie title is not in watched list 
+def watch_movie(user_data, title):
+    for movie_dict in user_data['watchlist']:
+        if title in movie_dict.values():
+            user_data['watchlist'].remove(movie_dict)
+            user_data['watched'].append(movie_dict)
+            return user_data
+    return user_data  
+
+# *****wave 2 testing test 10 - 13 ***** 
+# test 10 funct for avg rating that takes in user_data returns avg of ratings 
+
+# # test 11 function returns 0.0 if no movies in watched list
+def get_watched_avg_rating(user_data): 
+    if len(user_data['watched']) == 0:
+        mean = 0.0
+    else:
+        ratings = [movie['rating'] for movie in user_data['watched']]       
+        mean = sum(ratings) / len(ratings)
+    return mean 
+
+
+# test 12 funct collects the genre data and returns most watched genre 
+
+# test 13 make sure the function returns none if list is empty 
+def get_most_watched_genre(user_data):
+    word_count = {}
+    if user_data['watched'] == []:
+        return None
+    else:
+        for movie_dict in user_data['watched']:
+            if movie_dict['genre'] not in word_count:
+                word_count[movie_dict['genre']] = 1
+            else:
+                word_count[movie_dict['genre']] += 1
+        max_genre = max(word_count)
+        return max_genre
+
+
+
+#   ***** wave 3 tests 14-18 **** 
+# test 14 funct get_unique_watched takes in user data 
+# returns list of movies in user data that are not in friends data
+
+# test 15 returns an empty list if everyone watched the same movies 
+
+def get_unique_watched(user_data):
+
+    friends_watched = []
+    for dictionary in user_data['friends']:
+        for title_list in dictionary.values():
+            for dict in title_list:
+                friends_watched.append(dict)
+
+
+    unique_watched = [movie for movie in user_data['watched']\
+         if movie not in friends_watched]
+
+    # for movie_title_dict in user_data['watched']:
+    #     if movie_title_dict not in friends_watched:
+    #         unique_watched.append(movie_title_dict)
+
+    return unique_watched
+
+
+
+
+
+# test 16 function get_friends_unique_watched
+# takes in user_data determines which movies at least one of 
+# users friend has watched, but user has not.  return as 
+# list of dict. 
+
+# test 17 does it work with only one friend
+
+
+#  test 18 make sure the function returns [] if the user has 
+# watched all the movies friends have watched. 
+
+
+def get_friends_unique_watched(user_data): 
+    friends_watched = []
+# creates a list of dict of movies friends have watched 
+    for friend_dict in user_data['friends']:
+        for title_list in friend_dict.values():
+            for title in title_list:
+                friends_watched.append(title)
```

Comment:
Why not only append the movie if it's not in friends_watched already?  It might save you a loop.